src/mlpack/methods/ann/brnn_impl.hpp

```diff
@@ -83,6 +83,15 @@ double BRNN<OutputLayerType, MergeLayerType, MergeOutputType,
     ResetParameters();
   }
 
+  if (optimizer.MaxIterations() < this->predictors.n_cols)
+  {
+    Log::Warn << "The number of iterations is less than the size of the "
+              << "dataset, the optimizer might not pass over the entire "
+              << "dataset. Please modify the number of iterations to be "
+              << "at least equal to the number of columns of your "
+              << "dataset." << std::endl;
```

Comment:
Thanks for the clear and comprehensive warning message, I appreciate it.  It will be helpful for users.  I would suggest a few very simple changes though for clarity:

 - `The number of iterations` --> `The optimizer's maximum number of iterations`  (i.e. clarify that we're talking about the optimizer, and the maximum number of iterations)
 - `Please modify the number of iterations` -> `To fix this, modify the maximum number of iterations` ('Please modify' seems to suggest that the user *should* modify the maximum number of iterations, but technically they don't need to.  We just need to tell them how to fix the warning if they like.)
 - `number of columns` -> `number of points` (this one is really pedantic, but if you are using mlpack's bindings to other languages, then in those languages, a single point might correspond to a *row*, not a *column*; this is the case, for instance, in both the Julia bindings and the Python bindings.  So saying `column` is pretty specific to C++, but this warning might appear in other contexts, so we should make sure it's generic)

Reply:
@rcurtin, how about printing minimum required iterations ?

Reply:
Ahh, that would be nice, e.g., `at least equal to the number of points in the dataset (10042).` (or whatever number)